satpy/readers/grib.py

```diff
@@ -185,9 +188,10 @@ def _area_def_from_msg(self, msg):
             # take the corner points only
             lons = lons[([0, 0, -1, -1], [0, -1, 0, -1])]
             lats = lats[([0, 0, -1, -1], [0, -1, 0, -1])]
-            # correct for longitudes over 180
-            lons[lons > 180] -= 360
-
+            # if we have longitudes over 180, assume 0-360
+            if (lons > 180).any():
```

Comment:
In the case this is dask-array-based, this will trigger a computation right?

Reply:
This is being read from pygrib so it will never be a dask array.